Userland/Libraries/LibJS/Bytecode/Op.h

```diff
@@ -254,13 +254,13 @@ class NewArray final : public Instruction {
     {
     }
 
-    NewArray(Operand dst, AK::Array<Operand, 2> const& elements_range)
+    NewArray(Operand dst, ReadonlySpan<ScopedOperand> elements)
         : Instruction(Type::NewArray)
         , m_dst(dst)
-        , m_element_count(elements_range[1].index() - elements_range[0].index() + 1)
+        , m_element_count(elements.size())
     {
-        m_elements[0] = elements_range[0];
-        m_elements[1] = elements_range[1];
+        for (size_t i = 0; i < m_element_count; ++i)
+            m_elements[i] = elements[i];
```

Comment:
I'm not 100% if this was tried before, but IIRC there was some stuff about constructing big arrays making the register range version nicer, but I might be misremembering

Reply:
idk, it makes the NewArray instruction itself smaller, but at the cost of needing a huge setup sequence with guaranteed-sequential registers